packages/contracts-bedrock/contracts/L1/OptimismPortal.sol

```diff
@@ -111,25 +123,74 @@ contract OptimismPortal is Initializable, ResourceMetering, Semver {
     event WithdrawalFinalized(bytes32 indexed withdrawalHash, bool success);
 
     /**
-     * @custom:semver 1.0.0
+     * @notice Emitted when the pause is triggered.
+     *
+     * @param account Address of the account triggering the pause.
+     */
+    event Paused(address account);
+
+    /**
+     * @notice Emitted when the pause is lifted.
+     *
+     * @param account Address of the account triggering the unpause.
+     */
+    event Unpaused(address account);
+
+    /**
+     * @notice Reverts when paused.
+     */
+    modifier whenNotPaused() {
```

Comment:
I like that we're not adding more deps on OZ here, but should we hide this within a utility contract? Or did you implement it within the portal so that the `paused` var is appended to the layout rather than prepended?

Reply:
I went back and forth about making a `Pausable` contract and just decided to not do it. Happy to refactor

Reply:
Slight preference for external since it can be cleanly contained.

Reply:
The problem here is that it will push the storage layout down and since this is running in prod we can't do that nicely

Reply:
We do have [48 slots available](https://github.com/ethereum-optimism/optimism/pull/4921/files/b90a5d5f6e49e8d7e6c2230a7744097eacf41a15#diff-28810b593e6112074b564a947d3e4cc048acacdc3120da28b90081a405df4e41R62) though.

Reply:
I got over this.